### ISSUE TITLE
Add 'description' property to block patterns

### ISSUE THREAD

**Issue by enriquesanchez** [WordPress/gutenberg#23042]
Bock patterns could really benefit from a description property that helps describe what the pattern is. This would be particularly beneficial for users using assistive technology.

Props to @afercia for [providing this feedback and suggestion in Slack](https://wordpress.slack.com/archives/C02QB2JS7/p1591199298132000).

Block patterns currently consist of the following properties:

```
<?php
/**
 * Hero Two Columns block pattern.
 *
 * @package gutenberg
 */

return array(
	'title'         => __( 'Features and Services', 'gutenberg' ),
	'content'       => "...",
	'viewportWidth' => 1000,
);
```

I imagine we could add an additional property, called 'description':

```
<?php
/**
 * Hero Two Columns block pattern.
 *
 * @package gutenberg
 */

return array(
	'title'         => __( 'Features and Services', 'gutenberg' ),
        'description'   => "Three columns, each with one featured image, text and a call to action button",
	'content'       => "...",
	'viewportWidth' => 1000,
);
```
This description field could then be used to describe the pattern via an `aria-describedby` on an adjacent element that is visually hidden but available to assistive technologies.

**Comment by enriquesanchez** [comment]
>Should these also be searchable? Required?

I think they should be required.

>Also, in the example above, there is a 'viewportWidth' => 1000, that I can not find any documentation for?

Not sure about documentation, I believe it's a recent addition. But in short, that property allows you to specify basic proportions to how you want the pattern to be displayed in the inserter. Particularly useful for patterns that are very tall and look odd whe displayed in the inserter menu.

**Comment by audrasjb** [comment]
Hi, I agree it would be nice to milestone this for WP 5.5, if feasible :)

**Comment by enriquesanchez** [comment]
It'll be great to have this included with 5.5 and updated to reflect the changes in https://github.com/WordPress/gutenberg/pull/23608.

Anything left to be done before we can merge?
